ivadomed/loader/utils.py

```diff
@@ -637,15 +780,23 @@ def create_bids_dataframe(loader_params, derivatives):
         [prefix_fnames.append(s.split('.')[0]) for s in subject_files]
         deriv = df[df['path'].str.contains('derivatives')]['filename'].tolist()
         has_deriv = []
-        for p in prefix_fnames:
-            available = [d for d in deriv if p in d]
-            if available:
-                has_deriv.append(p)
-                for t in target_suffix:
-                    if t not in str(available):
-                        logger.warning("Missing target_suffix {} for subject {}.".format(t, p))
-            else:
-                logger.warning("Missing derivatives for subject {}. Skipping subject.".format(p))
+        if roi_suffix is not None:
+            for p in prefix_fnames:
+                available = [d for d in deriv if p in d]
+                if roi_suffix in ('|'.join(available)):
+                    has_deriv.append(p)
+                else:
+                    logger.warning("Missing ROI derivatives for subject {}. Skipping subject.".format(p))
+        else:
+            for p in prefix_fnames:
+                available = [d for d in deriv if p in d]
```

Comment:
I did a test on my local station on MRI data (data_example_spinegeneric), and even though derivatives is set to True, the df don't store the path to the derivatives, so at this line `deriv` is empty. I get the following error: `RuntimeError: Derivatives not found.` Is it normal and will be implemented in future PR or maybe it is a bug from pybids?

Reply:
It is supposed to work. Do you have another warning about a missing dataset_description.json file? Because of pybids, the file is mandatory in the "derivatives" folder. See note [here](https://github.com/ivadomed/ivadomed/blob/c65ab5af2ef33619c09f686355606ca941dacdaf/dev/df_new_loader.py#L28).

Reply:
Ah you're right this was it! There are no dataset_description.json file in derivatives (or labels) in our testing data. In a future PR, we should probably add a step: if there is no dataset_description.json in derivatives, but there is one in the main bids path, then simply copy it there. Most of our bids dataset don't have dataset_description at this location, so this would avoid the error.

Reply:
Good idea, I'll add it when dealing with the refactoring of the BIDSDataframe class (#595).